numpyro/examples/covtype.py

```diff
@@ -0,0 +1,91 @@
+import argparse
+import time
+
+import numpy as onp
+from sklearn.datasets import fetch_covtype
+
+import jax
+import jax.numpy as np
+from jax import random
+
+import numpyro.distributions as dist
+from numpyro.handlers import sample
+from numpyro.hmc_util import initialize_model
+from numpyro.mcmc import hmc
+from numpyro.util import tscan
+
+
+
+
+
+# TODO: add to datasets.py so as to avoid dependency on scikit-learn
+def load_dataset():
+    data = fetch_covtype()
+    features = data.data
+    labels = data.target
+
+    # normalize features and add intercept
+    features = (features - features.mean(0)) / features.std(0)
+    features = np.hstack([features, np.ones((features.shape[0], 1))])
+
+    # make binary feature
+    _, counts = onp.unique(labels, return_counts=True)
+    specific_category = np.argmax(counts)
+    labels = (labels == specific_category)
+
+    N, dim = features.shape
+    print("Data shape:", features.shape)
+    print("Label distribution: {} has label 1, {} has label 0"
+          .format(labels.sum(), N - labels.sum()))
+    return features, labels
+
+
+def model(data, labels):
+    N, dim = data.shape
+    coefs = sample('coefs', dist.norm(np.zeros(dim), np.ones(dim)))
+    logits = np.dot(data, coefs)
+    return sample('obs', dist.bernoulli(logits, is_logits=True), obs=labels)
+
+
+def benchmark_hmc(args, features, labels):
+    N, dim = features.shape
+    step_size = np.sqrt(0.5 / N)
+    trajectory_length = step_size * args.num_steps
+    init_params = {'coefs': random.normal(key=random.PRNGKey(0), shape=(dim,))}
+
+    _, potential_fn, _ = initialize_model(random.PRNGKey(1), model, (features, labels,), {})
+    init_kernel, sample_kernel = hmc(potential_fn, algo=args.algo)
+    t0 = time.time()
+    # Do not run warmup for consistent benchmark
+    hmc_state, _, _ = init_kernel(init_params, num_warmup_steps=0, step_size=step_size,
+                                  trajectory_length=trajectory_length, run_warmup=False)
+    t1 = time.time()
+    print("time for hmc_init: ", t1 - t0)
+
+    def transform(state): return {'coefs': state.z['coefs'],
+                                  'num_steps': state.num_steps}
+
+    def body_fn(state, i):
+        return sample_kernel(state)
+
+    hmc_state = hmc_state.update(step_size=step_size)
```

Comment:
Could you add a comment into the script for why this step is needed so we can address it in later PR? I guess the reason is we convert step_size to float in `hmc` init implementation?

Reply:
Actually the only reason why I had this was to run for 10 steps per iteration in HMC, but since we don't have the `num_steps` arg anymore and trajectory_length is already set, this was the simplest way to ensure that.

Reply:
I see. The other way is to set `adapt_step_size=False` so step_size will be constant during init_kernel. (more precisely, find_initial_step_size won't trigger.)

Reply:
Done.